include/power_grid_model/auxiliary/meta_data.hpp

```diff
@@ -169,23 +177,25 @@ struct MetaData {
     size_t alignment;
     std::vector<DataAttribute> attributes;
 
-    DataAttribute const& find_attr(std::string const& attr_name) const {
-        for (DataAttribute const& attr : attributes) {
-            if (attr.name == attr_name) {
-                return attr;
-            }
+    DataAttribute const& get_attr(std::string const& attr_name) const {
+        Idx const found = find_attr(attr_name);
+        if (found >= 0) {
+            return attributes[found];
         }
         throw UnknownAttributeName{attr_name};
     }
 
-    bool has_attr(std::string const& attr_name) const {
-        try {
-            find_attr(attr_name);
-        }
-        catch (const UnknownAttributeName&) {
-            return false;
+    Idx find_attr(std::string const& attr_name) const {
```

Comment:
Nice refactor. Do we want to use the minimum value of Idx instead of -1, or do you find that redundant, as an actual Idx should always be positive?

Reply:
I think -1 should be enough here.